mmdet/core/general_data/general_data.py

```diff
@@ -0,0 +1,498 @@
+    """
+
+    def __init__(self, meta_info=None, data=None):
+
+        self._meta_info_fields = set()
+        self._data_fields = set()
+
+        if meta_info is not None:
+            self.set_meta_info(meta_info=meta_info)
+        if data is not None:
+            self.set_data(data)
+
+    def set_meta_info(self, meta_info=None):
+        """Add meta information.
+
+        Args:
+            meta_info (dict): A dict contains the meta information
+                of image. such as `img_shape`, `scale_factor`, etc.
+                Default: None.
+        """
+        assert isinstance(meta_info,
+                          dict), f'meta should be a `dict` but get {meta_info}'
+        meta = copy.deepcopy(meta_info)
+        for k, v in meta.items():
+            # should be consistent with original meta_info
+            if k in self._meta_info_fields:
+                ori_value = getattr(self, k)
+                if isinstance(ori_value, (torch.Tensor, np.ndarray)):
+                    if (ori_value == v).all():
+                        continue
+                    else:
+                        raise KeyError(
+                            f'img_meta_info {k} has been set as '
+                            f'{getattr(self, k)} before, which is immutable ')
+                elif ori_value == v:
+                    continue
+                else:
+                    raise KeyError(
+                        f'img_meta_info {k} has been set as '
+                        f'{getattr(self, k)} before, which is immutable ')
+            else:
+                self._meta_info_fields.add(k)
+                self.__dict__[k] = v
+
+    def set_data(self, data=None):
+        """Update a dict to `data_fields`.
+
+        Args:
+            data (dict): A dict contains annotations of image or
+                model predictions. Default: None.
+        """
+        assert isinstance(data,
+                          dict), f'meta should be a `dict` but get {data}'
+        for k, v in data.items():
+            self.__setattr__(k, v)
+
+    def new_results(self, meta_info=None, data=None):
+        """Return a new results with same image meta information.
+
+        Args:
+            meta_info (dict): A dict contains the meta information
+                of image. such as `img_shape`, `scale_factor`, etc.
+                Default: None.
+            data (dict): A dict contains annotations of image or
+                model predictions. Default: None.
+        """
+        new_results = self.__class__()
+        new_results.set_meta_info(dict(self.meta_items()))
+        if meta_info is not None:
+            new_results.set_meta_info(meta_info)
+        if data is not None:
+            new_results.set_data(data)
+        return new_results
+
+    def keys(self):
+        """
+        Returns:
+            list: Contains all keys in data_fields.
+        """
+        return [key for key in self._data_fields]
+
+    def meta_keys(self):
```

Comment:
`meta_key` or `meta_info_key`?

Reply:
I have renamed all the prefix from `meta` to `meta_info`